### PR TITLE
Stacked and grouped bars

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Replaces #575 

Changes since the previous PR:

 + transposeCsv is not spreadCsv and does not incorporate stacking
 + offsetBar has become groupedBar. Also, the implementation has been updated so that the offset for each bar is no longer applied via decoration (which feels a bit cleaner).

### CODE REVIEW

File: src/util/rebind.js [hunk 1]
```diff
@@ -34,6 +34,8 @@ function capitalizeFirstLetter(str) {
  * be rebound.
  */
 export function rebindAll(target, source, prefix, exclusions) {
+    prefix = typeof prefix !== 'undefined' ? prefix : '';
```
Comment (ColinEberhardt) [inline src/util/rebind.js]: This doesn't look right - I don't think I have touched rebind. Might be a bad merge.

I'll look into this.

File: src/data/spreadCsv.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+
+// the D3 CSV loader / parser converts each row into an object with property names
+// derived from the headings in the CSV. The spreadCsv component converts this into an
+// array of series, one per column
+export default function spreadCsv() {
+
+    var xValueKey = '';
+    var yValue = function(row, key) {
+        // D3 CSV returns all keys as strings, this converts them to numbers
```
Comment (chrisprice) [inline src/data/spreadCsv.js]: keys = values?

File: src/data/spreadCsv.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                return key !== xValueKey;
+            })
+            .map(function(key) {
+                var series = data.map(function(row) {
+                    return {
+                        x: row[xValueKey],
+                        y: yValue(row, key)
+                    };
+                });
+                series.name = key;
```
Comment (chrisprice) [inline src/data/spreadCsv.js]: `series.key`?

File: src/data/spreadCsv.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+
+// the D3 CSV loader / parser converts each row into an object with property names
+// derived from the headings in the CSV. The spreadCsv component converts this into an
+// array of series, one per column
+export default function spreadCsv() {
```
Comment (chrisprice) [inline src/data/spreadCsv.js]: Just call it `spread`?

File: src/series/groupedBar.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import _bar from './bar';
```
Comment (chrisprice) [inline src/series/groupedBar.js]: `barSeries`

File: src/series/groupedBar.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import d3 from 'd3';
+import _dataJoin from '../util/dataJoin';
```
Comment (chrisprice) [inline src/series/groupedBar.js]: `dataJoinUtil`

File: src/util/rebind.js (hunk 1 above)
Comment (ColinEberhardt) [inline src/util/rebind.js]: No, not a bad merge, just poor memory. This allows properties to be re-bound without a prefix, as tested here:

https://github.com/ScottLogic/d3fc/pull/600/files#diff-f6d6160b813bdc9d086071b5af5c3e5eR73

**Comment by ColinEberhardt** [comment]
@chrisprice - thanks for the feedback. I've made those few naming tweaks and rebased onto master.

**Pull request merged**